File: versus/js/01-lambdas/consuming.js
function sendEmail(to, from, subject, message,
        success_cb, failure_cb) {
    // pretend we tried to send an email and call the
    // success or failure callback randomly.
    if (Math.random() > 0.5) {
        success_cb(to);
    } else {
        failure_cb(to);
    }
}

sendEmail('you@your_domain.com', 'me@my_domain.com', 
    'Very Important Email',
    'TODO: remember to write email body. :-D ',
    (to) => { 
        console.log('successful email sent to: ' + to);
    },
    (to) => {
        console.log('OH NO! Very important email not sent to ' + to);
    });

File: versus/go/01-lambdas/consuming.go
package main

import (
	"fmt"
	"math/rand"
)

func sendEmail(to string, from string, subject string,
               message string,
               successCb func(string),
               failureCb func(string)) {
	if rand.Float32() >= 0.5 {
		successCb(to)
	} else {
		failureCb(to)
	}
}

func main() {
	sendEmail("you@your_domain.com", "me@my_domain.com",
		"Very Important Email",
		"TODO: remember to write email body. :-D",
		func(to string) {
			fmt.Printf("Successful email sent to: %s\n", to)
		},
		func(to string) {
			fmt.Printf(
				"OH NO! Very important email not sent to %s\n",
				to)
		})

}
